llama-index-core/llama_index/core/indices/prompt_helper.py

```diff
@@ -11,12 +11,16 @@
 import logging
 from copy import deepcopy
 from string import Formatter
-from typing import Callable, List, Optional, Sequence
+from typing import TYPE_CHECKING, Callable, List, Optional, Sequence
+
+if TYPE_CHECKING:
+    from llama_index.core.tools import BaseTool
 
 from llama_index.core.base.llms.types import ChatMessage, LLMMetadata
 from llama_index.core.bridge.pydantic import Field, PrivateAttr
 from llama_index.core.constants import DEFAULT_CONTEXT_WINDOW, DEFAULT_NUM_OUTPUTS
 from llama_index.core.llms.llm import LLM
+from llama_index.core.llms.structured_llm import StructuredLLM
```

Comment:
i'm pretty sure this needs to go into type checking too, but maybe you can verify?

Reply:
Unit tests pass 👀 Its already import `core.llms.llm` before this too, which is under the same namespace